mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp

```diff
@@ -635,18 +677,114 @@ struct ConcatSliceOptimization : public OpRewritePattern<tosa::SliceOp> {
       sliceStart[axis] -= inputType.getDimSize(axis);
     }
 
-    if (!replaceWithSlice)
+    if (replaceWithSlice) {
+      rewriter.replaceOp(sliceOp, replaceWithSlice.value());
+      return success();
+    }
+
+    // If its not possible to replace the slice with an input to the concat, try
+    // to reduce the number of inputs to the concat
+    if (!concatOp->hasOneUse()) {
+      return rewriter.notifyMatchFailure(
+          sliceOp,
+          "Preceding concat must have a single use"); // Do not introduce new
+                                                      // concats
+    }
+    sliceStart.assign(sliceOp.getStart().begin(), sliceOp.getStart().end());
+    llvm::SmallVector<Value> concatsToSlice;
```

Comment:
I don't understand this variable name. This seems to be `newConcatOperands`?

Reply:
Yes, it was named badly, I renamed it now